python/pyarrow/parquet.py

```diff
@@ -1116,6 +1111,11 @@ class ParquetDataset:
     dataset metadata. Increasing this is helpful to read partitioned
     datasets.
 {0}
+use_legacy_dataset : bool, default True
```

Comment:
default False

Reply:
This is the docstring of ParquetDataset, for which this PR still keeps the old default (it's only for `read_table` that it is switching, as here there is only minimal impact on the API, as it simply returns a table. While for ParquetDataset this would not be the case)

Reply:
I see thanks